lib/gui/lvgl/lvgl.c

```diff
@@ -69,6 +69,19 @@ static void lvgl_log(lv_log_level_t level, const char *file, uint32_t line,
 
 static int lvgl_allocate_rendering_buffers(lv_disp_drv_t *disp_drv)
 {
+	struct display_capabilities cap;
+	struct device *display_dev = (struct device *)disp_drv->user_data;
+
+	display_get_capabilities(display_dev, &cap);
+
+	__ASSERT(cap.x_resolution <= CONFIG_LVGL_HOR_RES,
```

Comment:
These assertions could be turned into build time assertions if all displays use DT (not for now, just suggesting)